---
title: "An orphan bioinformatician"
date: 2017-05-02T23:09:10
draft: false
categories: ["Uncategorized"]
---

Not a post with code or plots today, but some worries. I'm a PhD student with a Masters degree in Biochemistry and biotechnology, and my (bio)informatics training is almost non-existent. And yet, that's what I do for more than 3/4 of my time in my PhD research.

Mick Watson introduced the concept of a [pet bioinformatician](http://www.opiniomics.org/a-guide-for-the-lonely-bioinformatician/), "a single bioinformatician employed within laboratory based group". However, since I'm not a full-grown bioinformatician I consider myself an orphan bioinformatician. The department and research group in which I work focusses mainly on genetics and (functional) wet lab work, with two central bioinformaticians handling most of the requirements for the researchers with standardised pipelines. As such the average scientist has a limited understanding of the "bioinformatic black box" which happens between handling the samples to the sequencing core and looking at variant tables. RNA-seq goes from samples directly to differentially expressed genes and deregulated pathways. Those two bioinformaticians are great and I can often ask their help, but they obviously lack the time to fully supervise my work. And they like different programming languages.

I started with a "typical genetics" project, but quickly I felt the call of the dark side and started focussing on bioinformatics. I've followed some excellent courses from the [VIB Bioinformatics Core](https://www.bits.vib.be/) to get me started with coding, but that's obviously not sufficient for learning good practices. While the internet is great for finding bits of code and asking questions, I guess having more appropriate supervision would be beneficial for my development. I'm looking for recommendations on how to overcome this issues. And very specifically, if someone would have the time to review my code (mainly Python, rarely R), that would be great.